python/shotgun_desktop/upgrade_startup.py

```diff
@@ -106,9 +84,20 @@ def upgrade_startup(splash, sgtk, app_bootstrap):
     out_of_date = (latest_descriptor.get_version() != current_desc.get_version())
 
     if not out_of_date:
-        logger.debug("Desktop startup does not need upgrading. Currenty running version %s" % current_desc.get_version())
+        logger.debug(
+            "Desktop startup does not need upgrading. Currenty running version %s" % current_desc.get_version()
+        )
+        return False
+
+    # FIXME: Violating internal API
+    from sgtk.commands.console_utils import _check_constraints
```

Comment:
yep exposing this - for example alongside https://github.com/shotgunsoftware/tk-core/blob/master/python/tank/descriptor/descriptor_bundle.py#L33 - sounds great to me!